test/jdk/javax/swing/JFileChooser/CustomApproveButtonTest.java

```diff
@@ -59,41 +60,48 @@ public void run() {
 
     private CustomApproveButtonTest(String lookAndFeel) {
         System.out.println("Testing Look & Feel : " + lookAndFeel);
-        frame = new JFrame("CustomApproveButtonTest");
+
+        setLookAndFeel(lookAndFeel);
 
         try {
-            UIManager.setLookAndFeel(lookAndFeel);
-        } catch (Exception e) {
-            throw new RuntimeException("Failed to set " + lookAndFeel +
-                    " LookAndFeel. Error : " + e);
-        }
+            frame = new JFrame("CustomApproveButtonTest");
 
-        JFileChooser fileChooser = new JFileChooser();
-        fileChooser.setDialogType(JFileChooser.CUSTOM_DIALOG);
-        frame.add(fileChooser);
-        frame.setVisible(true);
-        frame.pack();
+            JFileChooser fileChooser = new JFileChooser();
+            fileChooser.setDialogType(JFileChooser.CUSTOM_DIALOG);
+            frame.add(fileChooser);
+            frame.pack();
+            frame.setVisible(true);
 
-        JButton customApproveButton = fileChooser.getUI().getDefaultButton(fileChooser);
 
-        if (customApproveButton == null) {
-            fail("Cannot find Custom Approve Button in FileChooser!");
-            return;
-        }
-        if (customApproveButton.getText() == null) {
-            fail("Custom Approve Button Text is null in FileChooser!");
-            return;
-        }
+            JButton customApproveButton = fileChooser.getUI().getDefaultButton(fileChooser);
 
-        if (frame != null) {
-            frame.dispose();
+            if (customApproveButton == null) {
+                fail("Cannot find Custom Approve Button in FileChooser!");
+                return;
```

Comment:
```suggestion
                throw new RuntimeException("Cannot find Approve button in FileChooser!");
```
It's just an Approve button.

Now that frame is disposed of separately, there's no need for `fail` method — throw the exception.

Reply:
Updated.